src/secrets.ts

```diff
@@ -0,0 +1,58 @@
+import { HDKey } from '@scure/bip32';
+import { generateMnemonic, validateMnemonic, mnemonicToSeedSync } from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+import { encodeBase64toUint8 } from './base64';
+import { bytesToNumber } from './utils';
+import { hexToNumber } from '@noble/curves/abstract/utils';
+export const generateNewMnemonic = (): string => {
+	const mnemonic = generateMnemonic(wordlist, 128);
+	if (!validateMnemonic(mnemonic, wordlist)) {
+		return generateNewMnemonic();
+	}
```

Comment:
Why this recursive behaviour?

Reply:
If this really is a weird edge case, then probably better to do this in a while loop?

Reply:
looking at this again, i think this is entirely unnecessary, and the `generateMnemonic` function will always return a valid mnemonic.